src/main/cpp/mechanisms/noteManager/decoratormods/PassState.cpp

```diff
@@ -50,7 +50,10 @@ void PassState::Init()
 void PassState::Run()
 {
 	// Logger::GetLogger()->LogData(LOGGER_LEVEL::PRINT, string("ArrivedAt"), string("PassState"), string("run"));
-	bool angleIsWithinTolerance = abs(m_mechanism->getlauncherAngle()->GetCounts() - m_targetAngle) <= 0.5;
+	m_mechanism->SetLauncherAngleTarget(units::angle::degree_t(m_targetAngle));
+	m_mechanism->UpdateLauncherAngleTarget();
```

Comment:
UpdateLauncherAngleTarget() needs to be called in RunCommonTasks() as opposed to in some of the states. Otherwise as soon as you switch to a state in which UpdateLauncherAngleTarget() is not called, the PID calculations stop and the output from the roborio to the motor is stuck at whatever it was in the previous state

Reply:
We will go to the protected position through 	
if (protectLauncher)
	{
		SetLauncherToProtectedPosition();
	}

This way you only need to call UpdateLauncherAngleTarget() in the functions you are truly updating the launcher angle.

Do you agree?